core/src/program/blocks/span_block.rs

```diff
@@ -340,6 +334,18 @@ fn batch_ops(ops: Vec<Operation>) -> (Vec<OpBatch>, Digest) {
     (batches, hash)
 }
 
+/// Returns a vector of operations present in the input `span_vec` which is a vector of Span Blocks
+/// each containing vector of operation batches.  
+pub fn combine_spans(span_vec: &Vec<Span>) -> Vec<Operation> {
+    let mut ops = Vec::<Operation>::new();
+    for span in (*span_vec).iter() {
+        for batch in &span.op_batches {
+            ops.extend_from_slice(&batch.ops);
+        }
+    }
+    ops
```

Comment:
I was thinking this function would return `CodeBlock` (rather than just a vector of operations). And then here you could handle things like checking if there is only a single block in the vector, building the actual block, and maybe even clearing the input vector.

Reply:
Makes sense. I didn't have this in mind because then we didn't have a function which returns a `CodeBlock` from a Span. Now it makes sense to handle this check if there is only a single span block in the vector inside of it.